cmd/webhook/main.go

```diff
@@ -44,50 +59,60 @@ var (
 	}
 )
 
-var validationHook cmd.ValidatingAdmissionHook = handlers.NewFuncBackedValidator(logs.Log, GroupName, webhook.Scheme, validationFuncs)
-var mutationHook cmd.MutatingAdmissionHook = handlers.NewSchemeBackedDefaulter(logs.Log, GroupName, webhook.Scheme)
+var validationHook handlers.ValidatingAdmissionHook = handlers.NewFuncBackedValidator(logs.Log, GroupName, webhook.Scheme, validationFuncs)
+var mutationHook handlers.MutatingAdmissionHook = handlers.NewSchemeBackedDefaulter(logs.Log, GroupName, webhook.Scheme)
 
 func main() {
-	// Avoid "logging before flag.Parse" errors from glog
-	flag.CommandLine.Parse([]string{})
-
-	// parse the command line flags to pull out the tls-cert-file
-	// argument. This flag will be parsed by code inside cmd.RunAdmissionServer
-	// so no need to pass it through the call stack or have nice errors
-	tlsflagSet := flag.NewFlagSet("tls", flag.ContinueOnError)
-	tlsflagVal := tlsflagSet.String("tls-cert-file", "", "")
-	tlsflagSet.Parse(os.Args[1:])
-	if *tlsflagVal != "" {
-		runfilewatch(*tlsflagVal)
+	klog.InitFlags(flag.CommandLine)
+	flag.Parse()
+
+	log := klogr.New()
+	stopCh := setupSignalHandler()
+
+	var source server.CertificateSource
+	if tlsCertFile == "" || tlsKeyFile == "" {
+		log.Info("serving insecurely as tls certificate data not provided")
```

Comment:
Make this warning?

Reply:
Logr uses V levels instead of explicit Warn/Error/Info levels.This is being logged at V(0), the 'highest priority' (aside from .Error, which is a special case). I could prepend this with `warning: ` though?

Reply:
done